src/opt.rs

```diff
@@ -1,49 +1,50 @@
+use bpaf::*;
 use std::path::PathBuf;
-use structopt::StructOpt;
 
 use crate::choice::Choice;
-use crate::parse;
+use crate::parse::output_field_separator;
 
-#[derive(Debug, StructOpt)]
-#[structopt(name = "choose", about = "`choose` sections from each line of files")]
-#[structopt(setting = structopt::clap::AppSettings::AllowLeadingHyphen)]
+#[derive(Debug, Clone, Bpaf)]
+/// `choose` sections from each line of files
+#[bpaf(options, generate(options))]
 pub struct Opt {
     /// Choose fields by character number
-    #[structopt(short, long)]
+    #[bpaf(short, long)]
     pub character_wise: bool,
 
     /// Activate debug mode
-    #[structopt(short, long)]
+    #[bpaf(short, long)]
     pub debug: bool,
 
     /// Use exclusive ranges, similar to array indexing in many programming languages
-    #[structopt(short = "x", long)]
+    #[bpaf(short('x'), long)]
     pub exclusive: bool,
 
     /// Specify field separator other than whitespace, using Rust `regex` syntax
-    #[structopt(short, long)]
+    #[bpaf(short, long)]
     pub field_separator: Option<String>,
 
     /// Input file
-    #[structopt(short, long, parse(from_os_str))]
+    #[bpaf(short, long, argument("INPUT"), optional)]
     pub input: Option<PathBuf>,
 
     /// Use non-greedy field separators
-    #[structopt(short, long)]
+    #[bpaf(short, long)]
     pub non_greedy: bool,
 
     /// Index from 1 instead of 0
-    #[structopt(long)]
+    #[bpaf(long)]
     pub one_indexed: bool,
 
     /// Specify output field separator
-    #[structopt(short, long, parse(from_str = parse::output_field_separator))]
+    #[bpaf(short, long, argument("SEP"), map(output_field_separator), optional)]
     pub output_field_separator: Option<String>,
 
     /// Fields to print. Either a, a:b, a..b, or a..=b, where a and b are integers. The beginning
     /// or end of a range can be omitted, resulting in including the beginning or end of the line,
     /// respectively. a:b is inclusive of b (unless overridden by -x). a..b is
     /// exclusive of b and a..=b is inclusive of b.
-    #[structopt(required = true, min_values = 1, parse(try_from_str = parse::choice))]
+    #[bpaf(any::<Choice>("CHOICE"), some("At least one is required"))]
+    // parse(try_from_str = parse::choice))]
```

Comment:
Am I correct to assume we don't need this commented line anymore?

Reply:
Fixed.